app/views/hostgroups/_form.html.erb

```diff
@@ -27,10 +27,19 @@
       <%= text_f f, :name %>
       <%= textarea_f f, :description, :rows => :auto %>
       <%= hostgroup_puppet_environment_field f %>
-      <%= select_f(f, :compute_profile_id, ComputeProfile.visibles, :id, :name,
+
+      <%= select_f f, :compute_resource_id, accessible_resource(@hostgroup, :compute_resource), :id, :to_label,
+          { :include_blank => blank_or_inherit_f(f, :compute_resource, :inherited_blank_value => _('Bare Metal'), :blank_value => _('Bare Metal')) },
+          { :label => _('Deploy on'),
+            :onchange => 'tfm.hostgroups.computeResourceChanged(this);',
+            :help_inline => :indicator } if SETTINGS[:unattended] %>
+
+      <div id="compute_profile" <%= display?(!@hostgroup.compute_resource_id) %> >
```

Comment:
Yes, but you can achieve it with hiding the field as well: 

```
large_servers
  - deploy on : aws
  - compute profile: large

large_servers/aws
  - deploy on: aws (Inherit parent)
  - compute profile: large (Inherit parent)

large_servers/vmware
  - deploy on: vmware
  - compute profile: large (Inherit parent)
```
Hiding the field prevents you from having:
```
large_servers
  - deploy on: Bare metal
  - compute profile: large
```

but I guess this is just a matter of taste, so I made changes as suggested.